draco/analysis/beamform.py

```diff
@@ -0,0 +1,515 @@
+        Total transit time will be ~ 2 * timetrack.
+    """
+
+    collapse_ha = config.Property(proptype=bool, default=True)
+    polarization = config.Property(proptype=str, default='full')
+    timetrack = config.Property(proptype=float, default=900.)
+
+    def setup(self, manager):
+        """ Generic setup method. To be complemented by specific
+            setup methods in daughter tasks.
+
+        Parameters
+        ----------
+        manager : either `ProductManager`, `BeamTransfer` or `TransitTelescope`
+            Contains a TransitTelescope object describing the telescope.
+        """
+        # Get the TransitTelescope object
+        self.telescope = io.get_telescope(manager)
+        # Polarizations.
+        if self.polarization == 'I':
+            self.process_pol = ['XX', 'YY']
+            self.return_pol = ['I']
+        elif self.polarization == 'full':
+            self.process_pol = ['XX', 'YY', 'XY', 'YX']
+            self.return_pol = self.process_pol
+        elif self.polarization == 'copol':
+            self.process_pol = ['XX', 'YY']
+            self.return_pol = self.process_pol
+        elif self.polarization == 'stokes':
+            self.process_pol = ['XX', 'YY', 'XY', 'YX']
+            self.return_pol = ['I', 'Q', 'U', 'V']
+            msg = 'Stokes parameters are not implemented'
+            raise RuntimeError(msg)
+        else:
+            msg = 'Invalid polarization parameter: {0}'
+            msg = msg.format(self.polarization)
+            raise ValueError(msg)
+        # Number of polarizations to process
+        self.npol = len(self.process_pol)
+        self.latitude = np.deg2rad(ephemeris.CHIMELATITUDE)
+
+    def process(self):
+        """ Generic process method. Performs all the beamforming,
+            but not the data parsing. To be complemented by specific
+            process methods in daughter tasks.
+
+        Returns
+        -------
+        formed_beam : `containers.FormedBeam` or `containers.FormedBeamHA`
+            Formed beams at each source. Shape depends on parameter
+            `collapse_ha`.
+        """
+        # Contruct containers for formed beams
+        if self.collapse_ha:
+            # Container to hold the formed beams
+            formed_beam = containers.FormedBeam(
+                    freq=self.freq,
+                    object_id=self.source_cat.index_map['object_id'],
+                    pol=np.array(self.return_pol))
+        else:
+            # Container to hold the formed beams
+            formed_beam = containers.FormedBeamHA(
+                    freq=self.freq, ha=np.arange(self.nha, dtype=int),
+                    object_id=self.source_cat.index_map['object_id'],
+                    pol=np.array(self.return_pol))
+            # Initialize container to zeros.
+            formed_beam.fbeam[:] = 0.
+            formed_beam.ha[:] = 0.  # TODO: Should this be np.nan instead?
```

Comment:
I'd imagine it should be zero. You should typically never introduce a NaN yourself. What's the argument it should be NaN?

If you're worried about how to take care of missing data, you should probably have a weight dataset.